esphome/components/online_image/__init__.py

```diff
@@ -0,0 +1,161 @@
+import logging
+from esphome import automation
+import esphome.config_validation as cv
+import esphome.codegen as cg
+from esphome.components.http_request import (
+    CONF_HTTP_REQUEST_ID,
+    HttpRequestComponent,
+)
+from esphome.components.image import (
+    Image_,
+    IMAGE_TYPE,
+    CONF_USE_TRANSPARENCY,
+    validate_cross_dependencies,
+)
+from esphome.const import (
+    CONF_BUFFER_SIZE,
+    CONF_FORMAT,
+    CONF_ID,
+    CONF_TRIGGER_ID,
+    CONF_TYPE,
+    CONF_RESIZE,
+    CONF_URL,
+)
+
+DEPENDENCIES = ["display"]
+AUTO_LOAD = ["http_request", "image"]
+CODEOWNERS = ["@guillempages"]
+MULTI_CONF = True
+
+CONF_ON_DOWNLOAD_FINISHED = "on_download_finished"
+CONF_ON_ERROR = "on_error"
+
+_LOGGER = logging.getLogger(__name__)
+
+online_image_ns = cg.esphome_ns.namespace("online_image")
+
+ImageFormat = online_image_ns.enum("ImageFormat")
+IMAGE_FORMAT = {"PNG": ImageFormat.PNG}  # Add new supported formats here
+
+OnlineImage = online_image_ns.class_("OnlineImage", cg.PollingComponent, Image_)
+
+# Actions
+SetUrlAction = online_image_ns.class_(
+    "OnlineImageSetUrlAction", automation.Action, cg.Parented.template(OnlineImage)
+)
+ReleaseImageAction = online_image_ns.class_(
+    "OnlineImageReleaseAction", automation.Action, cg.Parented.template(OnlineImage)
+)
+
+# Triggers
+DownloadFinishedTrigger = online_image_ns.class_(
+    "DownloadFinishedTrigger", automation.Trigger.template()
+)
+DownloadErrorTrigger = online_image_ns.class_(
+    "DownloadErrorTrigger", automation.Trigger.template()
+)
+
+ONLINE_IMAGE_SCHEMA = cv.Schema(
+    {
+        cv.Required(CONF_ID): cv.declare_id(OnlineImage),
+        cv.GenerateID(CONF_HTTP_REQUEST_ID): cv.use_id(HttpRequestComponent),
+        #
+        # Common image options
+        #
+        cv.Optional(CONF_RESIZE): cv.dimensions,
+        cv.Optional(CONF_TYPE, default="BINARY"): cv.enum(IMAGE_TYPE, upper=True),
+        # Not setting default here on purpose; the default depends on the image type,
+        # and thus will be set in the "validate_cross_dependencies" validator.
+        cv.Optional(CONF_USE_TRANSPARENCY): cv.boolean,
+        #
+        # Online Image specific options
+        #
+        cv.Required(CONF_URL): cv.url,
+        cv.Optional(CONF_FORMAT, default="PNG"): cv.enum(IMAGE_FORMAT, upper=True),
```

Comment:
If the intention is to add more formats in the future, then this should not have a default even if there is only one option. Otherwise it would be a breaking change in the future to remove the default.
```suggestion
        cv.Required(CONF_FORMAT): cv.enum(IMAGE_FORMAT, upper=True),
```

Reply:
The idea was to also add an "AUTO" format, that would choose depending on the Content-Type returned by the server; and have that as default later on.
But since at the moment I'm not sure if the http_request component can be used to get extra headers, it's probably better to make `format` mandatory as you suggest, and it can always be made optional again later on without being a breaking change.